pkg/engine/virtualmachinescalesets.go

```diff
@@ -577,11 +580,17 @@ func CreateAgentVMSS(cs *api.ContainerService, profile *api.AgentPoolProfile) Vi
 		}
 		vmssStorageProfile.DataDisks = getVMSSDataDisks(profile)
 	} else {
-		imageRef := profile.ImageRef
-		useAgentCustomImage := imageRef != nil && len(imageRef.Name) > 0 && len(imageRef.ResourceGroup) > 0
-		if useAgentCustomImage {
-			vmssStorageProfile.ImageReference = &compute.ImageReference{
-				ID: to.StringPtr(fmt.Sprintf("[resourceId(variables('%[1]sosImageResourceGroup'), 'Microsoft.Compute/images', variables('%[1]sosImageName'))]", profile.Name)),
+		if profile.HasImageRef() {
+			imageRef := profile.ImageRef
```

Comment:
these new lines don't have UT coverage, see https://github.com/Azure/aks-engine/blob/master/pkg/engine/virtualmachinescalesets_test.go#L701 for an example of a VMSS unit test.